### ISSUE TITLE
Blurred border around windows

### ISSUE THREAD

**Issue by orangebowlerhat** [tryone144/compton#45]
Ubuntu Desktop 19.10, Xfce 4.14

NVidia GTX 670, nvidia driver 440, one monitor

Latest version of kawase, without alpha hack
```
# Shadow
shadow = true;
no-dnd-shadow = true;
no-dock-shadow = true;
clear-shadow = true;
shadow-radius = 12;
shadow-offset-x = -10;
shadow-offset-y = -7;
shadow-opacity = 0.7;
# shadow-red = 0.0;
# shadow-green = 0.0;
# shadow-blue = 0.0;
shadow-exclude = [
	"name = 'Notification'",
	"class_g = 'Conky'",
	"class_g ?= 'Notify-osd'",
	"class_g = 'Cairo-clock'",
	"_GTK_FRAME_EXTENTS@:c"
];

#shadow-exclude = "n:e:Menu";
# shadow-exclude-reg = "x10+0+0";
# xinerama-shadow-crop = true;

# Opacity
menu-opacity = 0.8;
#menu-opacity-override = false;
#inactive-opacity = 0.8;
#active-opacity = 0.8;
active-opacity-override = true;
frame-opacity = 0.8;
inactive-opacity-override = false;
alpha-step = 0.06;
#inactive-dim = 0.2;
#inactive-dim-fixed = true;
#blur-background = true;
blur-background-frame = true;
blur-method = "kawase";
blur-strength = 10;
blur-background-fixed = false;
#blur-kern = "11x11gaussian";
#blur-kern = "5,5,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1";

blur-background-exclude = [
	"window_type = 'dock'",
	"window_type = 'desktop'"
];

#"_NET_WM_OPAQUE_REGION@:c",
#"_GTK_FRAME_EXTENTS@:c",
#opacity-rule = [ "80:class_g = 'URxvt'" ];

# Fading
fading = true;
# fade-delta = 30;
fade-in-step = 0.06;
fade-out-step = 0.04;
#no-fading-openclose = true;
#no-fading-destroyed-argb = true;
fade-exclude = [ ];

# Other
backend = "glx";
mark-wmwin-focused = true;
mark-ovredir-focused = true;
# use-ewmh-active-win = true;
detect-rounded-corners = true;
detect-client-opacity = true;
refresh-rate = 0;
vsync = "opengl-swc";
dbe = false;
paint-on-overlay = true;
# sw-opti = true;
# unredir-if-possible = true;
# unredir-if-possible-delay = 5000;
# unredir-if-possible-exclude = [ ];
focus-exclude = ["class_g = 'Cairo-clock'"];
detect-transient = true;
detect-client-leader = true;
invert-color-include = [ ];
# resize-damage = 1;

# GLX backend
#glx-no-stencil = false;
glx-copy-from-front = false;
# glx-use-copysubbuffermesa = true;
glx-no-rebind-pixmap = true;
glx-swap-method = "undefined";
# glx-use-gpushader4 = true;
# xrender-sync = true;
# xrender-sync-fence = true;

# Window type settings
wintypes:
{
 	tooltip = {fade = true; shadow = false; opacity = 0.75; focus = true; };
};
```

Many windows have a weird blurred transparent border around them. This is an example of a menu in Galculator. Most menus and tooltips have them, some top level windows also have it around the window. The width of the blurred margin seems to depends on what type of window it is.

![image](https://user-images.githubusercontent.com/12849146/80873570-806dc900-8cb1-11ea-969d-653742c153b5.png)

I have tried disabling shadows in comptom.conf, I have tried removing shadows from the GTK theme I'm using (Qogir-dark) and I have tried changing blur exclude. However, none of these have fixed the problem so far. I have tried several settings for blur exclude. GTK_FRAME_EXTENTS, removes the blur from many windows entirely. _NET_WM_OPAQUE_REGION produces the same result as I have now.

My best guess is that it relates to the WM drawing shadows around the windows, specifically because the width of the border changes. I don't know if its the theme, the decorator, or compton drawing the shadows. None of them seem to make any difference. I cannot figure out how to make the blur stay inside the window, or turn these shadows off.

**Comment by tryone144** [comment]
I suspect the window manager / theme is doing something here. This looks oddly similar to the problems other client-side decorated windows cause (like firefox menus).

If you enable shadows in compton, are these drawn around the actual window or around the blurred area? I currently have no setup to try and reproduce this specific issue.

**Comment by orangebowlerhat** [comment]
Thanks for replying @tryone144 The problem is to do with CSD in the theme itself. I've removed most of the shadows from its gtk.css and the result is much better.

![image](https://user-images.githubusercontent.com/12849146/81104368-4ba87e80-8f0a-11ea-915d-8620055e4253.png)

I first tried nocsd-gtk3 but that didn't seem to get all of them, particularly not the menus. There's still a couple of windows with very small blurred margins around them but I suspect I can figure that out now that I know what the problem is.

Still, its looking beautiful now. Thanks for making it.
